src/interpreter/mod.rs

```diff
@@ -176,19 +177,19 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         use rustc_const_math::{ConstInt, ConstIsize, ConstUsize, ConstFloat};
         macro_rules! i2p {
             ($i:ident, $n:expr) => {{
-                let ptr = self.memory.allocate($n)?;
+                let ptr = self.memory.allocate($n, $n)?;
```

Comment:
This over-aligns `u64` on linux, and might underalign on some hypothetical system where the alignment of primitives is bigger than the size, but if that happens, the write will fail, so let's be lazy here :)